src/wp-includes/meta.php

```diff
@@ -1264,6 +1330,37 @@ function register_meta( $object_type, $meta_key, $args, $deprecated = null ) {
 		}
 	}
 
+	if ( array_key_exists( 'default', $args ) ) {
+		if ( false === $args['single'] && ! wp_is_numeric_array( $args['default'] ) ) {
+			$args['default'] = array( $args['default'] );
```

Comment:
Can we drop this upgrading? This doesn't work if the user is using an `array` default, which seems like a footgun. I think it'd be better to always require passing an array of values when using a non-single meta key.

Reply:
There is an example of this mistake in the tests: https://github.com/WordPress/wordpress-develop/pull/256/files#diff-427a6864697f781b7b497562637fbeb1R619

Reply:
Example of that footgun: https://github.com/WordPress/wordpress-develop/pull/256/files#diff-427a6864697f781b7b497562637fbeb1R778

Reply:
And another on line 800.